src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -521,7 +562,21 @@ def sql_query(self) -> str:
         if not self.has_clause and not self.projection:
             self._sql_query = self.from_.sql_query
             return self._sql_query
-        from_clause = self.from_.sql_in_subquery
+        placeholder = f"{analyzer_utils.LEFT_PARENTHESIS}{self.from_._id}{analyzer_utils.RIGHT_PARENTHESIS}"
```

Comment:
As of the discussion offline, for risk control, better use the new parameter to protect the old code so the old code path remains as before when the param is not enabled.
Similarly please use the param to protect wherever is needed throughout this PR.

Reply:
done. It seems the only place requiring param protection in this PR (all others are adding functions/attributes)